infrastructure/bin/deploy.sh

```diff
@@ -7,12 +7,41 @@ set -o pipefail
 # Fail if using undefined variables
 set -o nounset
 
-if [ -z "$1" ]; then
+Help()
+{
+  echo "Deploy the project from Azure Pipelines."
+  echo
+  echo "Syntax: deploy.sh [-h] [-d] [-r <ROOT_DIR>]"
+  echo "   -h   Show this help message."
+  echo "   -d   Setup the environment with dev dependencies."
+  echo "   -r   Absolute path to the root directory."
+  echo
+}
+
+GCDT_DEV=false
+
+while getopts "hdr:" option; do
+  case $option in
+    h) # display Help
+      Help
+      exit;;
+    d) # setup for dev
```

Comment:
My only suggestion is to use longform options, `--dev` and `--root` so that its a bit more obvious when reading the yml files

Reply:
Unfortunately, it seems that `getops` doesn't support long form options.  :cry: 
https://stackoverflow.com/a/402410
I added some comments to the yml file, though, to try to make it clearer.
@petertgiles
[yml comments for options](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9301/commits/fa67d082908d2f825842b0c046a5475c532468d4)